crates/bevy_reflect/src/serde/de/deserializer.rs

```diff
@@ -93,28 +284,57 @@ use serde::de::{DeserializeSeed, Error, IgnoredAny, MapAccess, Visitor};
 /// [`Box<DynamicList>`]: crate::DynamicList
 /// [`FromReflect`]: crate::FromReflect
 /// [`ReflectFromReflect`]: crate::ReflectFromReflect
-pub struct ReflectDeserializer<'a> {
+/// [`with_processor`]: Self::with_processor
+pub struct ReflectDeserializer<'a, P = ()> {
```

Comment:
I think we should bound `P` here to help prevent cases where users accidentally attempt to use the wrong type, and wonder why they can't deserialize anything.
```suggestion
pub struct ReflectDeserializer<'a, P: ReflectDeserializerProcessor = ()> {
```

Reply:
Makes sense to me, typically you wouldn't put bounds on the struct itself but I think it's fine here